addons/mail/static/tests/owl/component/chat_window_manager_tests.js

```diff
@@ -601,6 +601,423 @@ QUnit.test('chat window: open / close', async function (assert) {
+        .click();
+    await testUtils.nextTick(); // re-render
+
+    // SETUP INITIAL STATE
+    // - Set a scroll position to chat window
+    document.querySelector(`.o_MessageList.o_Thread_messageList`).scrollTop = 142;
+
+    // - Set html content of the composer of the chat window
+    const composerTextInput = document.querySelector(`
+        .o_ComposerTextInput
+        > .note-editor
+        > .note-editing-area
+        > .note-editable`);
+    composerTextInput.innerHTML = "XDU for the win !";
+    composerTextInput.dispatchEvent(new window.KeyboardEvent('input'));
+
+    // - Set attachments of the composer
+    const composerLocalId = Object.keys(self.store.state.composers)[0];
+    const composerAttachmentLocalIds = [
+        this.store.dispatch('createAttachment', {
+            filename: 'blah.png',
+            id: 1,
+            name: 'blah',
+        }),
+        this.store.dispatch('createAttachment', {
+            filename: 'blah.pdf',
+            id: 2,
+            name: 'bluh',
+        }),
+    ];
+    this.store.dispatch('linkAttachmentToComposer', composerLocalId, composerAttachmentLocalIds[0]);
+    this.store.dispatch('linkAttachmentToComposer', composerLocalId, composerAttachmentLocalIds[1]);
+    await testUtils.nextTick();
+
+    // TESTS
+
+    // - phase 1 : on hiding menu
+    // -- Hide home menu
+    await this.widget.call('chat_window', 'test:will_hide_home_menu');
+    await this.widget.call('chat_window', 'test:hide_home_menu');
+    let chatWindowStates;
+
+    // -- Check store state
+    chatWindowStates = this.store.state.chatWindowManager.storedChatWindowStates;
+    assert.ok(chatWindowStates, "chat window manager should have storedChatWindowState key in store");
+    assert.ok(chatWindowStates['mail.channel_20'], "chat window state should be stored");
+    assert.strictEqual(chatWindowStates['mail.channel_20'].scrollTop, 142, "chat window scrollTop value saved in store should be good");
+    assert.strictEqual(chatWindowStates['mail.channel_20'].composerTextInputHtmlContent, "XDU for the win !", "chat window composer text input html saved in store should be good");
+    assert.deepEqual(chatWindowStates['mail.channel_20'].composerAttachmentLocalIds, composerAttachmentLocalIds, "chat window composer attachments saved in store should be good");
+
+    // - phase 2 : on showing menu
+    // Show home menu
+    await this.widget.call('chat_window', 'test:will_show_home_menu');
+    await this.widget.call('chat_window', 'test:show_home_menu');
+
+    // -- Check store state
+    chatWindowStates = this.store.state.chatWindowManager.storedChatWindowStates;
+    assert.ok(chatWindowStates, "chat window manager should have storedChatWindowState key in store");
+    assert.ok(chatWindowStates['mail.channel_20'], "chat window state should be stored");
+    assert.strictEqual(chatWindowStates['mail.channel_20'].scrollTop, 142, "chat window scrollTop value saved in store should be good");
```

Comment:
If this test fails, it logs
```
chat window scrollTop value saved in store should be good
Expected: 142
Result: 950
```

Hard to tell at a glance what's the purpose of this assertion, why 142 is a good value and not 800, etc.
_"Chat window thread scroll positions should be the same as before showing home menu"_ better catches the intent of this assertion.